l3build-arguments.lua

```diff
@@ -252,8 +253,22 @@ local function argparse()
          end
         end
       else
-        stderr:write("Unknown option " .. a .."\n")
-        return { target = "help" }
+        -- Private special debugging options "--debug-<key>"
+        local key = match(a, "^%-%-debug%-(%w[%w%d_]*)")
+        if key then
+          ---@type l3build_t
+          local l3build = require("l3build")
+          l3build.debug[key] = true
+          i = i + 1
+          if i <= #arg then
+            goto top
+          else
+            break
+          end
+        else
```

Comment:
This is where the "--debug-..." is implemented.

This will be refactored with many simplifications and no goto.